networkx/algorithms/tsp.py

```diff
@@ -0,0 +1,299 @@
+"""
+================================
+Travelling Salesman Problem (TSP)
+================================
+
+Implementation of heuristic and metaheuristic algorithms
+for solving and approximating the TSP problem.
+
+Categories of algorithms which are implemented:
+- Greedy
+- Simulated Annealing (SA)
+
+Travelling Salesman Problem tries to find, given the weight
+(distance) between all points where salesman has to visit, the
+route so that:
+- Total distance (cost) which salesman travels to be minimized.
+- Salesman has to return to the point where he stated.
+- Salesman has to visit each point only once apart fom source
+point.
+
+It is an NP-hard problem in combinatorial optimization,
+important in operations research and theoretical computer science.
+
+http://en.wikipedia.org/wiki/Travelling_salesman_problem
+"""
+from __future__ import division
+import math
+from random import choice, randint, random
+import networkx as nx
+import itertools
```

Comment:
Imports should be in the order 1. built-in libraries, 2. third-party libraries, 3. this library, and then alphabetized within each block.